client/my-sites/stats/stats-purchase/stats-purchase-tier-upgrade-slider.tsx

```diff
@@ -29,16 +39,72 @@ function useTranslatedStrings() {
 	};
 }
 
-function TierUpgradeSlider( { className }: TierUpgradeSliderProps ) {
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const MOCK_PLAN_DATA = [
+	{
+		price: '$9',
+		views: 10000,
+		description: '$9/month for 10k views',
+	},
+	{
+		price: '$19',
+		views: 100000,
+		description: '$19/month for 100k views',
+	},
+	{
+		price: '$29',
+		views: 250000,
+		description: '$29/month for 250k views',
+	},
+	{
+		price: '$49',
+		views: 500000,
+		description: '$49/month for 500k views',
+	},
+	{
+		price: '$69',
+		views: 1000000,
+		description: '$69/month for 1M views',
+	},
+	{
+		price: '$89.99',
+		views: '1M++',
+		extension: true,
+		per_unit_fee: 1799,
+		description: '$25/month per million views if views exceed 1M',
+	},
+];
+
+function TierUpgradeSlider( { className, priceTiers, currencyCode }: TierUpgradeSliderProps ) {
+	const translate = useTranslate();
+	const infoReferenceElement = useRef( null );
 	const componentClassNames = classNames( 'stats-tier-upgrade-slider', className );
+	const EXTENSION_THRESHOLD = 2; // in millions
 
-	// Get the plan details.
-	const plans = useAvailableUpgradeTiers();
+	// Transform plan details to dusplay.
+	const plans = priceTiers?.map( ( plan ): StatsPlanTierUI => {
+		if ( plan?.maximum_units === null ) {
+			// highest tier extension
+			return {
+				price: plan?.minimum_price_monthly_display,
+				views: plan?.minimum_units,
+				extension: true,
+				per_unit_fee: plan?.per_unit_fee,
+			};
+		}
+
+		return {
+			price: plan?.maximum_price_monthly_display,
+			views: plan?.maximum_units,
+		};
+	} );
+
+	// const plans = MOCK_PLAN_DATA; // TODO: REMOVE MOCK DATA!
```

Comment:
I restored mock data, but it will be available locally - not through the live branch. This way even if we forget to remove the mock, the component will work as it needs to and we can test various currencies on the the live branch.